### PR TITLE
Bugfix dismissing r event while telegrab selected

### PR DESCRIPTION
**Pull request by RZME**
fixes dismissing the random event while having telegrab spell selected

### CODE REVIEW

**Comment by ChrisBPB** [comment]
you are editing RSBot files, not your project files, so your changes must be general use - not specific to you

In order for this to be effective you should add a getSelectedSpell() or similar method to Magic (mirror changes should be added to rt6).

Additionally clicking specific co-ordinates is bad for resizable, and in general just a bad idea because you don't want 100,000 users clicking the same spot every time a random event appears.

**Comment by ChrisBPB** [comment]
https://github.com/powerbot/powerbot/blob/rsbot-api/src/main/java/org/powerbot/script/rt4/Magic.java#L29


will already return the selected Spell

**Comment by RZME** [comment]
tried to find the widget component for the uptext to check but couldnt find it. But regarding the action that should be taken: will ctx.menu.click(menuCommand -> menuCommand.action.equals("Cancel")); work?

**Comment by ChrisBPB** [comment]
Probably not because inventory won't say cancel it would say Use or something.

At the current time, while we don't have a hook for the selected Spell, you would just for if(ctx.magic.spell()!=Spell.Nil){
//cancel the spell, but not my clicking coordinated
}

**Comment by ChrisBPB** [comment]
I was saying if an inventory item is selected it would be Use not Cancel, and they may be other cases but I don't know

I imagine your way would work though so put whatever you think is best and Big Coma can review it

**Comment by ChrisBPB** [comment]
formatting is messed up and probably not good idea for inline comments like that, the code itself is clean so there shouldn't be confusion from reading

also to add, on further thinking I probably would not rely on your Cancel menu thing because if the spell has runes the option will be Cast or something which still needs to be cancelled - how you handle it needs to be thought of carefully so you don't attack a random NPC, or alch a party hat - etc

**Comment by Strikegently** [comment]
You can click the inventory to cancel a spell cast.

**Comment by Strikegently** [comment]
What is
`ctx.input.click(ctx.widgets.widget(320).component(Random.nextInt(1,24)).component(0).centerPoint(),true);`

??

**Comment by RZME** [comment]
click in a random inventory box (widget.click didnt work for me -> tested it)

**Comment by ChrisBPB** [comment]
as mentioned previously you have to be careful you don't alch something that shouldn't be alched

**Comment by Strikegently** [comment]
1. If the user is already on the magic tab, then `ctx.game.tab(Game.Tab.MAGIC);` won't do anything.
1. You don't do anything in the event of failure to deselect the spell.
1. Fix the formatting.

**Comment by Strikegently** [comment]
1. Please fix the formatting. 
1. Use more descriptive commit messages. I'll squash your commits with a proper message, but please do it properly in the future.

File: src/main/java/org/powerbot/bot/rt4/RandomEvents.java [hunk 1]
```diff
@@ -34,7 +34,20 @@ public void poll() {
 		if (!threshold.contains(this)) {
 			threshold.add(this);
 		}
-		
+	        if(ctx.magic.spell()!= Magic.Spell.NIL){
```
Comment (Strikegently) [inline src/main/java/org/powerbot/bot/rt4/RandomEvents.java]: Fix the spacing.

File: src/main/java/org/powerbot/bot/rt4/RandomEvents.java [hunk 2]
```diff
@@ -34,7 +34,20 @@ public void poll() {
...
+		   	} else {
+	                	ctx.game.tab(Game.Tab.INVENTORY);
+		   	}
+			      
+	           	Condition.wait(new Condition.Check() {
+				@Override
+				public boolean poll() {
+			    		return ctx.magic.spell()== Magic.Spell.NIL;
+				}
+		    	},200,10);
```
Comment (Strikegently) [inline src/main/java/org/powerbot/bot/rt4/RandomEvents.java]: Spacing.

File: src/main/java/org/powerbot/bot/rt4/RandomEvents.java [hunk 3]
```diff
@@ -34,7 +34,20 @@ public void poll() {
...
+	        	if(!ctx.game.tab(Game.Tab.MAGIC)) {
+		       		ctx.game.tab(Game.Tab.MAGIC);
+		   	} else {
+	                	ctx.game.tab(Game.Tab.INVENTORY);
+		   	}
+			      
+	           	Condition.wait(new Condition.Check() {
+				@Override
+				public boolean poll() {
+			    		return ctx.magic.spell()== Magic.Spell.NIL;
```
Comment (Strikegently) [inline src/main/java/org/powerbot/bot/rt4/RandomEvents.java]: Spacing.

**Comment by Strikegently** [comment]
@RZME 
If you fix the formatting, we can get this merged.

**Comment by Strikegently** [comment]
Closing since he apparently doesn't see this.

**Pull request closed without merging**